pkg/liqoctl/status/peer/peerinfo.go

```diff
@@ -0,0 +1,352 @@
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package statuspeer
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/utils/strings/slices"
+
+	discoveryv1alpha1 "github.com/liqotech/liqo/apis/discovery/v1alpha1"
+	liqoconsts "github.com/liqotech/liqo/pkg/consts"
+	"github.com/liqotech/liqo/pkg/liqoctl/output"
+	"github.com/liqotech/liqo/pkg/liqoctl/status"
+	"github.com/liqotech/liqo/pkg/liqoctl/status/utils/resources"
+	liqonetutils "github.com/liqotech/liqo/pkg/liqonet/utils"
+	liqoutils "github.com/liqotech/liqo/pkg/utils"
+	liqogetters "github.com/liqotech/liqo/pkg/utils/getters"
+	liqolabels "github.com/liqotech/liqo/pkg/utils/labels"
+	peeringconditionsutils "github.com/liqotech/liqo/pkg/utils/peeringConditions"
+	"github.com/liqotech/liqo/pkg/utils/slice"
+)
+
+// PeerInfoOptions contains the options for the PeerInfoChecker.
+type PeerInfoOptions struct {
+	RemoteClusterNames []string
+}
+
+// PeerInfoChecker implements the Check interface.
+// holds the information about the peered cluster.
+type PeerInfoChecker struct {
+	o                *status.Options
+	pio              *PeerInfoOptions
+	peerInfoSection  output.Section
+	collectionErrors []status.CollectionError
+}
+
+const (
+	peerInfoCheckerName = "Peered Cluster Information"
+)
+
+// NewPeerInfoChecker return a new PeerInfoChecker.
+func NewPeerInfoChecker(o *status.Options, remoteClusterNames ...string) *PeerInfoChecker {
+	return &PeerInfoChecker{
+		peerInfoSection: output.NewRootSection(),
+		o:               o,
+		pio: &PeerInfoOptions{
+			RemoteClusterNames: remoteClusterNames,
+		},
+	}
+}
+
+// Silent implements the Check interface.
+func (pic *PeerInfoChecker) Silent() bool {
+	return false
+}
+
+// Collect implements the collect method of the Checker interface.
+// it collects the infos of the peered cluster.
+func (pic *PeerInfoChecker) Collect(ctx context.Context) {
+	localClusterIdentity, err := liqoutils.GetClusterIdentityWithControllerClient(ctx, pic.o.CRClient, pic.o.LiqoNamespace)
+	localClusterName := localClusterIdentity.ClusterName
+	if err != nil {
+		pic.addCollectionError("LocalClusterName", "", err)
+		return
+	}
+
+	foreignClusterList, err := liqogetters.ListForeignClustersByLabel(ctx, pic.o.CRClient, labels.Everything())
+	if err != nil {
+		pic.addCollectionError("ForeignClusters", "", err)
+		return
+	}
+
+	remoteClusterNamesNotFound := pic.checkPeerExistence(foreignClusterList)
```

Comment:
This is probably useless, since already retrieved by the function below (since the fc for non existing ones is constructed custom and it can be identified).